Issue-60

morganwootton opened the pull request:
# New content

<!--- Briefly describe the changes made. -->

<!--- Replace XXX with any associated issues. --->

Resolves #XXX

## Checklist

- [ ] New pages are added to the table of contents.
- [ ] Everything displays correctly in the web-generated preview.
- [ ] Appropriate attribution has been given to source materials.
- [ ] My name is included alphabetically in `credits.md`.

morganwootton commented:
How is my image? Should I try to make the background transparent? If so, how would I accomplish that?

mphoward commented:
I think it looks nice, and a white background is fine! You may want to try to crop it slightly to reduce the extra space around it. You can also specify its display width and alignment in your file by adding lines like:
```
:width: 300px
:align: center
```
to the `{image}` directive.

mphoward commented on first-order-odes/integrating-factor.md:
`````diff
@@ -1 +1,71 @@
-# Integrating factor
+# Integrating factor. 
+
+## 4. Example:
+Mole balance on a tank with increasing inlet/outlet flow rate.
+
+```{image} ./_images/Integrating_factors_image.jpg
+:alt: Tank with inlet and outlet flows and varying concentrations
+```
+
+\begin{align}
+\dot{n}_{A} &= \dot{n}_{A,\text{in}} - \dot{n}_{A,\text{out}}, \quad \dot{n}_{A,\text{in}} = C_{f,A} \dot{q}t \\
+\dot{n}_{A} &= C_{f,A}\dot{q}t - \frac{n_A}{V} \dot{q}t \quad \dot{n}_{A,\text{out}} = \frac{n_A}{V}\dot{q}t \\
+\end{align}
+
+````{dropdown} 4. Solution
+Rewrite: 
+```{math}
+\begin{align}
+\dot{n}_{A} + (\frac{\dot{q}t}{V}) n_{A} &= C_{f,A} \dot{q}t \\
+p(t) &= \frac{\dot{q}t}{V} \quad r(t) = C_{f,A}  \dot{q}t \\
+F(t) &= exp[\int{p(t)dt}] = exp(\int{\frac{\dot{q}t}{v}dt}) = exp(\frac{1}{2}\frac{\dot{q}}{v^{2}})\\
+\end{align}
+```
+
+```{math}
+\begin{align}
+\int{F(t)r(t)dt} &= \int{exp(\frac{1}{2}\frac{\dot{q}}{V}t^{2})C_{f,A}\dot{q}tdt} = \int{e^{u}C_{f,A}Vdu} = C_{f,A}Vexp(\frac{1}{2}\frac{\dot{q}}{v}t^{2}) \\
+u &= \frac{1}{2}\frac{\dot{q}}{v}t^{2}\\
+du &= \frac{\dot{q}}{V}tdt \to \dot{q}tdt = Vdu\\
+n_{A}(t) &= exp(-\frac{1}{2}\frac{\dot{q}}{V}t^{2})[C_{f,A}Vexp(\frac{1}{2}\frac{\dot{q}}{v}t^{2})+C]\\
+n_{A}(o) &= n_A,o = C_{f,A}V+C \to{C = n_{A,o} - C_{f,A}V}
+\end{align}
+```
+
+```{math}
+\to{\boxed{n_{A}(t) = C_{f,A}V + (n_{A,o} - C_{f,A}V)exp{(-\frac{1}{2}\frac{\dot{q}}{v}t^{2})}}}\\
+````
`````
Since these are examples, you can wrap them in `{example}`:

````````suggestion
````{example} Mole balance on a tank with increasing inlet/outlet flow rate.

```{image} ./_images/Integrating_factors_image.jpg
:alt: Tank with inlet and outlet flows and varying concentrations
```

\begin{align}
\dot{n}_{A} &= \dot{n}_{A,\text{in}} - \dot{n}_{A,\text{out}}, \quad \dot{n}_{A,\text{in}} = C_{f,A} \dot{q}t \\
\dot{n}_{A} &= C_{f,A}\dot{q}t - \frac{n_A}{V} \dot{q}t \quad \dot{n}_{A,\text{out}} = \frac{n_A}{V}\dot{q}t \\
\end{align}

---

Rewrite:

\begin{equation}
\dot{n}_{A} + (\frac{\dot{q}t}{V}) n_{A} &= C_{f,A} \dot{q}t
\end{equation}

This is a linear first-order ODE with
\begin{equation}
p(t) = \frac{\dot{q}t}{V}, \qquad r(t) = C_{f,A}  \dot{q}t
\end{equation}

First, compute the integrating factor *F*:

\begin{align}
F(t) &= exp[\int{p(t)dt}] = exp(\int{\frac{\dot{q}t}{v}dt}) \\
 &= exp(\frac{1}{2}\frac{\dot{q}}{v^{2}})
\end{align}

\begin{align}
\int{F(t)r(t)dt} &= \int{exp(\frac{1}{2}\frac{\dot{q}}{V}t^{2})C_{f,A}\dot{q}tdt} = \int{e^{u}C_{f,A}Vdu} = C_{f,A}Vexp(\frac{1}{2}\frac{\dot{q}}{v}t^{2}) \\
u &= \frac{1}{2}\frac{\dot{q}}{v}t^{2}\\
du &= \frac{\dot{q}}{V}tdt \to \dot{q}tdt = Vdu\\
n_{A}(t) &= exp(-\frac{1}{2}\frac{\dot{q}}{V}t^{2})[C_{f,A}Vexp(\frac{1}{2}\frac{\dot{q}}{v}t^{2})+C]\\
n_{A}(o) &= n_A,o = C_{f,A}V+C \to{C = n_{A,o} - C_{f,A}V}
\end{align}

\begin{equation}
n_{A}(t) = C_{f,A}V + (n_{A,o} - C_{f,A}V)exp{(-\frac{1}{2}\frac{\dot{q}}{v}t^{2})}
\end{equation}
````
````````

Could you also please add a few sentences to explain what is being done? I gave an example of how this might be done for your first block of equations. Can you do something similar for the second block, e.g., to separate evaluation of the integral, application of the initial condition, and then the final solution

mphoward commented on first-order-odes/integrating-factor.md:
`````diff
@@ -1 +1,71 @@
+\to{\boxed{n_{A}(t) = C_{f,A}V + (n_{A,o} - C_{f,A}V)exp{(-\frac{1}{2}\frac{\dot{q}}{v}t^{2})}}}\\
+````
+
+5. Example: Hormone level (again)
`````
Please also put this in an `{example}` and apply similar changes as above.

mphoward commented on first-order-odes/integrating-factor.md:
```diff
@@ -1 +1,71 @@
-# Integrating factor
+# Integrating factor. 
```
Revert:

```suggestion
# Integrating factor
```

mphoward reviewed (changes requested):
Thank you! Overall this looks really good. I have some suggested changes to improve formatting and readability. Can you please apply these, then request another review?

morganwootton commented:
I have added more commentary? Is this sufficient? Additionally when I added ```{Example} my work was no longer displaying as it was previously when I looked at my preview. Is there a way to fix this?

mphoward reviewed (changes requested):
Thank you! I fixed the issues you were having with getting these to display, and I also edited the formatting the first example. Could you do something similar to the second example, following the style of the first one? Please re-request another review when you're ready!

mphoward reviewed (approved):
This looks great! There were a couple formatting issues in the hormone example, but I cleaned these up for you. Thanks for your nice work on this!
